### ISSUE TITLE
How to use SinglePolicyInterpreter for new unseen data?

### ISSUE THREAD

**Issue by titubs** [microsoft/EconML#678]
@kant @Padarn @kbattocchi  Hello, my understanding is that one can use SinglePolicyInterpreter to apply to new instances (unseen data) in order to decide if a treatment shall be given. Is that true?/ If so, then, I wonder if SinglePolicyInterpreter can also be used for a continous treatment such as 0-10 for example?

I understand that the SinglePolicyInterpreter will decide what the best splits are but **can the SinglePolicyInterpreter build a policy around a continuous treatment?** If so, is there any example notebook please?

**Comment by kbattocchi** [comment]
Yes, although primarily designed for discrete treatments, SingleTreePolicyInterpreter can also be used with a continuous treatment; in that case the policy will effectively tell you if the marginal value of the treatment is positive - it will not tell you "how much" treatment to give because the assumption is that if the marginal value is positive then you should give as much treatment as you can, though you should be careful extrapolating too far.  See e.g. the "Make Policy Decisions with EconML" section of [this customer segmentation notebook](https://github.com/microsoft/EconML/blob/main/notebooks/CustomerScenarios/Case%20Study%20-%20Customer%20Segmentation%20at%20An%20Online%20Media%20Company.ipynb).

The emphasis of this class is on creating easily *interpretable* policies, in the form of a simple tree, given an existing CATE estimator; alternatively you could generate policies that are potentially more tailored to each individual unit (but less interpretable) using the CATE estimator directly.

**Comment by titubs** [comment]
@kbattocchi Here is the output of my policy. and as you mentioned, it is able to process continuous variables but the issue is that the policy does not tell me how much treatment to give as per screenshot (splits by treatment value). You mentioned other alternatives, could you share what those are as I need to make recommendations that segment such as:

1. Give a set of users between 0-5% discount
2. Give a set of users between 5-10% discount
3. Give no discount.

Please let me know what my options are. Thanks for all your help, I truly appreciate it.

<img width="707" alt="Screen Shot 2022-09-20 at 9 06 34 PM" src="https://user-images.githubusercontent.com/107452474/191412102-5c95787a-d795-4f1e-9e3a-a99ce5646586.png">

**Comment by kbattocchi** [comment]
Most of our estimators are assuming a linear treatment effect, so if marginal effect is positive (which is what the tree is showing) then continuing to increase the treatment intensity is beneficial.  This is probably unrealistic for something like a discount, for a few reasons: first of all, you are unlikely to have any data for extreme values of discount so you should not trust the model to extrapolate to these regions, and secondly, the effect is probably not approximately linear for large deviations.  It is also possible that you are not accounting for the costs of the discounts, but that depends on the nature of your outcome (if it's net profit, for example, then that should presumably include all costs, whereas just revenue could give a misleading impression if discounts lead to lots of revenue where goods are sold below cost, for example).

Beyond making sure you're correctly accounting for the cost of discounts, one way to deal with the linearity issue is to perform a low degree polynomial expansion of the discount and use that as the treatment instead of just the discount by itself, which would allow you to learn a non-linear treatment effect; this will not work correctly with the policy interpreters, though, which try to pick the "best" treatment which won't make sense if the treatments are not actually independent options.
